src/Utility/TextSplitter.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by dawid on 23.10.2020.
+//
+
+#include "TextSplitter.h"
+
+int TextSplitter::rangeWidth(long long int textLength) {
+    return ceil((double) textLength / (double) numOfThreads);
+}
+
+// It's hard cut the text into substrings of equal size without getting out of range sometimes,
+// so when we get out of range a <-1, -1> is pushed back to indicate that this tuple is wrong and thread should not process it
+
+std::vector<std::tuple<long long int, long long int>>& TextSplitter::splitText(long long int textLength, int patternLength) {
```

Comment:
Also we can use std::pair here.

Reply:
We can, or we should? Is there a big difference? Teach me c++ master :D

Reply:
We can. I think `std::pair` is cleaner. The difference in performance is non-existant. `std::touple` can be used to store more than 2 values. Std::pair is  always well... a pair :P It provides easier access to the variables by memeber functions `.first()` and `.second()`.  To access the values stored in `std::touple` you must use a friend function `get<>()`. Pair implementation is based on touple

Reply:
Okey, I think that std::pair looks more neat as well, I'll change that